components/browser/tabstray/src/main/java/mozilla/components/browser/tabstray/TabsAdapter.kt

```diff
@@ -43,7 +43,11 @@ class TabsAdapter(
     private var tabs: Tabs? = null
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TabViewHolder {
-        return viewHolderProvider.invoke(parent, tabsTray)
+        val result = viewHolderProvider.invoke(parent, tabsTray)
+        tabs?.let {
```

Comment:
`onCreateViewHolder` is a called every time a new tabs needs to be created, so this would be invoked n-tab times.

What we probably want is for an observer to know when the RecyclerView is done laying out the views so we can scroll to right one. I did some googling around on how others use this API:

What we can do is add the [`RecyclerView.AdapterDataObserver`](https://developer.android.com/reference/androidx/recyclerview/widget/RecyclerView.AdapterDataObserver) to know when the data has changed and we need to scroll to the right position.

This could be done in `BrowserTabsTray`.

Please also add tests, we have a guide on getting full code coverage to aid with testing too: https://mozac.org/contributing/code-coverage

Reply:
> What we probably want is for an observer to know when the RecyclerView is done laying out the views so we can scroll to right one. I did some googling around on how others use this API:
> 
> What we can do is add the RecyclerView.AdapterDataObserver to know when the data has changed and we need to scroll to the right position.

It looks like this is already setup for you in the tabs tray, does it not work as expected? https://github.com/mozilla-mobile/android-components/blob/master/components/feature/tabs/src/main/java/mozilla/components/feature/tabs/tabstray/TabsTrayPresenter.kt#L87